File: three/three.py
#!/usr/bin/env python

# The prime factors of 13195 are 5, 7, 13 and 29.
# What is the largest prime factor of the number 600851475143 ?

import os, sys
dirname, filename = os.path.split(os.path.abspath(sys.argv[0]))
util_dir = dirname+"/../util/"
sys.path.append(util_dir)

from util import factor

def main():
    mN=600851475143
    print [f for (f,x) in (factor(mN))][-1]

if __name__ == "__main__":
    main()

File: three/three.rb
#!/usr/bin/env ruby

# The prime factors of 13195 are 5, 7, 13 and 29.
# What is the largest prime factor of the number 600851475143 ?

UTIL="util"
UTIL_DIR="../#{UTIL}"

require File.join(File.dirname(__FILE__), "#{UTIL_DIR}", "#{UTIL}")
include Util

N = 600851475143

def main
    largest=Util::generate_factors(N)[-1]
    puts "#{largest}" # print largest # puts largest.to_s
end

if __FILE__==$0
    main()
end